### PR TITLE
OBSDEF-15149 Fix to remove tooltip from show/hide columns

### PR DESCRIPTION
**Pull request by m-manaswini**
This PR contains fix to remove tooltip from show/hide columns

image
image

### CODE REVIEW

File: src/datagrid/HideShowColumns.tsx [hunk 1]
```diff
@@ -216,7 +216,7 @@ export class HideShowColumns extends React.PureComponent<HideShowColumnsProps, H
                             <ul className={ClassNames.SWITCH_CONTENT}>
                                 {columns &&
                                     columns.map((column: DataGridColumn, index: number) => {
-                                        return this.buildColumnList(column, index);
+                                        return column.isHidden ? undefined : this.buildColumnList(column, index);
```
Comment (akate1) [inline src/datagrid/HideShowColumns.tsx:219]:
```suggestion
                                        return !column.isHidden && this.buildColumnList(column, index);
```

File: src/datagrid/DataGrid.tsx [hunk 2]
```diff
@@ -1146,6 +1152,8 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
                                     const showColumn =
                                         this.isDetailPaneOpen() &&
                                         column.columnID !== this.getFirstVisibleColumn()!.columnID
+                                            ? false
+                                            : column.isHidden
```
Comment (akate1) [inline src/datagrid/DataGrid.tsx:1156]: instate of this can we modify getFirstVisibleColumn() ?

Comment (m-manaswini) [reply src/datagrid/DataGrid.tsx:1154]: simplified this

File: src/datagrid/HideShowColumns.tsx (hunk 1 above)
Comment (m-manaswini) [reply src/datagrid/HideShowColumns.tsx:219]: done

File: src/datagrid/DataGrid.tsx [hunk 3]
```diff
@@ -870,7 +874,7 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
     // get details of first visible column
     private getFirstVisibleColumn = () => {
         const {allColumns} = this.state;
-        return allColumns.find((column: DataGridColumn) => column.isVisible === true);
+        return allColumns.find((column: DataGridColumn) => column.isVisible === true || !column.isHidden === true);
```
Comment (akate1) [inline src/datagrid/DataGrid.tsx:877]:
```suggestion
        return allColumns.find((column: DataGridColumn) => column.isVisible  || !column.isHidden );
```

**Review by akate1 — APPROVED** [review]
LGTM

**Review by psivakumaran — APPROVED** [review]

**Pull request merged**